src/Cli/dotnet/Commands/Test/MTP/TestApplication.cs

```diff
@@ -298,24 +307,17 @@ private void StoreOutputAndErrorData(Process process)
         process.BeginErrorReadLine();
     }
 
-    private bool ModulePathExists()
-    {
-        if (!File.Exists(Module.RunProperties.Command))
-        {
-            // TODO: The error should be shown to the user, not just logged to trace.
-            Logger.LogTrace($"Test module '{Module.RunProperties.Command}' not found. Build the test application before or run 'dotnet test'.");
-
-            return false;
-        }
-        return true;
-    }
-
     public void OnHandshakeMessage(HandshakeMessage handshakeMessage, bool gotSupportedVersion)
         => _handler.OnHandshakeReceived(handshakeMessage, gotSupportedVersion);
 
     private void OnCommandLineOptionMessages(CommandLineOptionMessages commandLineOptionMessages)
     {
-        HelpRequested?.Invoke(this, new HelpEventArgs { ModulePath = commandLineOptionMessages.ModulePath, CommandLineOptions = [.. commandLineOptionMessages.CommandLineOptionMessageList.Select(message => new CommandLineOption(message.Name, message.Description, message.IsHidden, message.IsBuiltIn))] });
+        if (!TestOptions.IsHelp)
```

Comment:
If user provides an invalid argument, then console shows error and prints all possible arguments. In that case `TestOptions.IsHelp` is false. Can you please check if this scenario is handled here correctly?

Reply:
If user provides an invalid argument, it will be forwarded to the test app as-is and it will fail. We don't print all possible arguments, and that's by-design.